referenceframe/frame.go

```diff
@@ -400,20 +399,88 @@ func (rf *rotationalFrame) Name() string {
 }
 
 func (rf *rotationalFrame) MarshalJSON() ([]byte, error) {
-	m := FrameMapConfig{
-		"type":    "rotational",
-		"name":    rf.name,
-		"rotAxis": rf.rotAxis,
-		"limit":   rf.limits,
-	}
-	return json.Marshal(m)
+	return json.Marshal(rf.toConfig())
+}
+
+func (rf *rotationalFrame) toConfig() FrameMapConfig {
+	config := rf.baseFrame.toConfig()
+	config["type"] = "rotational"
+	config["rotAxis"] = rf.rotAxis
+	return config
 }
 
 func (rf *rotationalFrame) AlmostEquals(otherFrame Frame) bool {
 	other, ok := otherFrame.(*rotationalFrame)
 	return ok && rf.baseFrame.AlmostEquals(other.baseFrame) && spatial.R3VectorAlmostEqual(rf.rotAxis, other.rotAxis, 1e-8)
 }
 
+type linearlyActuatedRotationalFrame struct {
```

Comment:
Ray, you're gonna hate me for this. 😉 

But we refer to the joint style elsewhere in `rdk` as Prismatic (see `motionplan/kinematic.go`, bunch of places here in `referenceframe`, etc.) but we're calling this a linearly actuated frame. Do we care that we're mixing terms?

Reply:
I feel like we need a little bit of a cleanup.

where "rotational" exists, "translational" should follow. These refer to frames.

where "revolute" exists, "prismatic" should follow. These refer to joints.

where "rotary" exists, "linear" should follow. These refer to actuators.

Reply:
The non-linearity describes a linear actuator changing a revolute joint. A little bit of a hmm moment for me.